daisy_workflows/image_build/debian/build.py

```diff
@@ -123,6 +124,50 @@ def main():
   synopsis_dest = os.path.join(outs_path, 'synopsis.json')
   utils.UploadFile('/tmp/synopsis.json', synopsis_dest)
 
+  # Create and upload metadata of the image and packages
+  logging.info('Creating image metadata.')
+  image = {
+      "family": "debian-9",
+      "name": re.match("debian-9-[\\w-]*", image_dest),
+      "version": "stretch",
+      "location": image_dest,
+      "release_date": datetime.now(timezone.utc),
+      "packages": []
+  }
+  # Read list of guest package
+  with open("guest_package") as f:
+    guest_packages = f.read().splitlines()
+
+  for package in guest_packages:
+    cmd = "dpkg-query -W --showformat '${Package}:${Version}:${Git}'" \
+          + package
+    code, stdout = utils.Excute(cmd, capture_output=True)
+    if code == 0:
+      splits = stdout.decode('utf-8').split(':')
+      package_name = splits[0]
+      package_version = splits[1]
+      package_git_url = splits[2]
+      start = package_version.index(":")
+      end = package_version.rindex(".")
+      package_release_date = package_version[start: end]
+      metadata = {
+          "name": package_name,
+          "version": package_version,
+          "commit_hash": package_git_url,
+          "release_date": package_release_date,
+          "stage": repo
+      }
+      image["packages"].append(metadata)
+
+    # Write image metadata to a file
+    with open('/tmp/metadata.json', 'w') as f:
+      f.write(json.dumps(image))
+
+    logging.info('Uploading image metadata.')
+    metadata_dest = os.path.join(image_dest.strip("root.tar.gz"),
```

Comment:
why don't we also pass in a `destination` variable for debian ? this approach makes debian somewhat less flexible than EL

Reply:
the image_dest is already existed compared to EL. I can make it align with EL to avoid a strip here. But a little bit tricky here is, we can reuse "destination" variable in "build" and "export_image" steps for EL. however, for debian, we need to keep both "image_dest" and "destination" together even they has the same gcs folder prefix. 
imag_dest: "gcs://path/to/folder/root.tar.gz"
destination: "gcs://path/to/folder/metadata.json"